src/task.ts

```diff
@@ -28,22 +28,26 @@ class TaskClient {
     const url = `tasks/${uid}`
     return await this.httpRequest.get<Task>(url)
   }
+
   /**
    * Get tasks
    *
-   * @param  {TaskParams} params - query parameters
+   * @param  {TaskParams} parameters - Parameters to browse the tasks
    *
-   * @returns { Promise<Result<Task[]>> }
+   * @returns {Promise<TasksResults>} - Promise containing all tasks
    */
-  async getTasks(params: TaskParams = {}): Promise<Result<Task[]>> {
+  async getTasks(parameters: TaskParams = {}): Promise<TasksResults> {
     const url = `tasks`
 
     const queryParams = {
-      indexUid: params?.indexUid?.join(','),
-      type: params?.type?.join(','),
-      status: params?.status?.join(','),
+      indexUid: parameters?.indexUid?.join(','),
```

Comment:
I'm not sure how you could implement that here, but in the dart I've created a class and I could extract all these params handling to a commonplace.

[lib/src/query_parameters/tasks_query.dart](https://github.com/meilisearch/meilisearch-dart/blob/1ab0a558639f69dc9a8d1947df2247e73cfc41d8/lib/src/query_parameters/tasks_query.dart)

Reply:
Since it only happens here is it worth it to go through a separate class?

Reply:
No, it's not a rule, but just something you can think about!